agents/random.py

```diff
@@ -20,13 +20,25 @@ def __init__(
     def forward(self, inputs: Dict) -> Dict:
         neighbors = inputs[self._neighbors_prefix]  # (batch_size, neighbors_cnt)
 
-        neighbors_mask = inputs[f'{self._neighbors_prefix}.mask']  # (batch_size, neighbors_cnt)
-        neighbors_mask_sum = torch.sum(neighbors_mask, dim=-1)  # (batch_size)
+        neighbors_length = neighbors.lengths
 
-        next_neighbors = torch.from_numpy(np.random.randint(
-            low=np.zeros_like(neighbors_mask_sum.cpu().detach().numpy()),
-            high=neighbors_mask_sum.cpu().detach().numpy()
-        )).to(neighbors.device)  # (batch_size)
+        low = torch.zeros_like(neighbors_length)
+        high = neighbors_length
+        next_neighbors = torch.randint(low=low.item(), high=high.item(), size=(1,))
```

Comment:
It is not usually `size=(1,)`
Remember that previously, this variable had the following shape: `# (batch_size)`

Btw, after these changes you don't need to create `low` and `high` anymore.

```suggestion
        next_neighbors = torch.round(torch.rand(neighbors_length.shape) * neighbors_length).long()  # (batch_size)
```